pkg/gateway/operations/putobject.go

```diff
@@ -74,11 +73,22 @@ func handleCopy(w http.ResponseWriter, req *http.Request, o *PathOperation, copy
 		return // operation already failed
 	}
 	ent.CreationDate = time.Now()
-	ent.Path = o.Path
-	err := o.Catalog.CreateEntry(req.Context(), o.Repository.Name, o.Reference, *ent)
+	address, err := o.Catalog.GenerateName()
 	if err != nil {
-		o.Log(req).WithError(err).Error("could not write copy destination")
-		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInvalidCopyDest))
+		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInternalError))
+		return
+	}
+	sourceObjectPointer := block.ObjectPointer{StorageNamespace: o.Repository.StorageNamespace, Identifier: ent.PhysicalAddress}
+	destinationObjectPointer := block.ObjectPointer{StorageNamespace: o.Repository.StorageNamespace, Identifier: address}
+
+	err = o.BlockStore.Copy(req.Context(), sourceObjectPointer, destinationObjectPointer)
+	if err != nil {
+		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInternalError))
+		return
+	}
+	err = o.finishUpload(req, ent.Checksum, address, ent.Size, true)
+	if err != nil {
+		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInternalError))
```

Comment:
I don't understand: what happened to the previous `InvalidCopyDest` error return?